packages/react/lib/ui/useThreeDSecure.ts

```diff
@@ -23,37 +23,56 @@ export function useThreeDSecure(opts?: UseThreeDSecureOptions) {
 
   const start = useCallback(
     (session: string, callbacks?: UseThreeDSecureCallbacks) => {
-      if (instance.current) return;
+      if (instance.current) {
+        instance.current.unmount();
+        instance.current = null;
+      }
 
       async function init() {
         const evervault = await ev;
         if (!evervault) return;
-        instance.current = evervault.ui.threeDSecure(session, opts);
+        try {
+          instance.current = evervault.ui.threeDSecure(session, opts);
 
-        if (callbacks?.onReady) {
-          instance.current.on("ready", callbacks.onReady);
-        }
+          if (callbacks?.onReady) {
+            instance.current.on("ready", callbacks.onReady);
+          }
 
-        if (callbacks?.onSuccess) {
-          instance.current.on("success", callbacks.onSuccess);
-        }
+          if (callbacks?.onSuccess) {
+            instance.current.on("success", callbacks.onSuccess);
+          }
 
-        if (callbacks?.onFailure) {
-          instance.current.on("failure", callbacks.onFailure);
-        }
+          if (callbacks?.onFailure) {
+            instance.current.on("failure", callbacks.onFailure);
+          }
 
-        if (callbacks?.onError) {
-          instance.current.on("error", callbacks.onError);
-        }
+          if (callbacks?.onError) {
+            instance.current.on("error", callbacks.onError);
+          }
 
-        instance.current.mount();
+          instance.current.mount();
+        } catch (error) {
+          instance.current = null;
```

Comment:
unmount needed here? if not, should we leave the session in current so a user can restart by manually unmounting/invoking start again?

Reply:
I think not needed yeah